src/RL/nn_structures/heterogeneousfullfeaturedcpnn.jl

```diff
@@ -21,36 +21,27 @@ Flux.@functor HeterogeneousFullFeaturedCPNN
 
 # TODO make it possible to use global features
 function (nn::HeterogeneousFullFeaturedCPNN)(states::BatchedHeterogeneousTrajectoryState)
-    @assert !isnothing(states.allValuesIdx)
-
     variableIdx = states.variableIdx
     batchSize = length(variableIdx)
-    allValuesIdx = states.allValuesIdx
-    actionSpaceSize = size(allValuesIdx, 1)
+    actionSpaceSize = size(states.fg.valnf, 2)
     mask = device(states) == Val(:gpu) ? CUDA.zeros(Float32, 1, actionSpaceSize, batchSize) : zeros(Float32, 1, actionSpaceSize, batchSize) # this mask will replace `reapeat` using broadcasted `+`
 
     # chain working on the graph(s) with the GNNs
     featuredGraph = nn.graphChain(states.fg)
-    nodeFeatures = featuredGraph.nf # FxNxB
-    variableFeature = fg.varnf[:, variableIdx]
-    valueFeatures = fg.valnf[:, possibleValuesIdx]
+    variableFeatures = featuredGraph.varnf # FxNxB
+    valueFeatures = featuredGraph.valnf
     globalFeatures = featuredGraph.gf # GxB
 
     # Extract the features corresponding to the varibales
     variableIndices = nothing
     Zygote.ignore() do
         variableIndices = Flux.unsqueeze(CartesianIndex.(variableIdx, 1:batchSize), 1)
```

Comment:
I know that this hasn't been reconsidered for a while but I don't understand why we have to transform the Cartesian Index into a 1-row matrix of type : `1×4 Matrix{CartesianIndex{2}}`.

Reply:
Well, it is just a matter of dimensions :

Here is a little dummy example done on a handcrafted `batchedDefaultTrajectoryState `: 
```julia
julia> idx = CartesianIndex.(zip([3,4], 1:2))
2-element Vector{CartesianIndex{2}}:
 CartesianIndex(3, 1)
 CartesianIndex(4, 2)

julia> idx_unsqueeze = Flux.unsqueeze(CartesianIndex.(zip([3,4], 1:2)),1)
1×2 Matrix{CartesianIndex{2}}:
 CartesianIndex(3, 1)  CartesianIndex(4, 2)

julia> batched_state_2.fg.nf[:,idx]
3×2 Matrix{Float32}:
 0.0  0.0
 1.0  1.0
 0.0  0.0

julia> batched_state_2.fg.nf[:,idx_unsqueeze]
3×1×2 Array{Float32, 3}:
[:, :, 1] =
 0.0
 1.0
 0.0

[:, :, 2] =
 0.0
 1.0
 0.0
```

Reply:
Thank you for this example!